heartbeat/ocf.py

```diff
@@ -141,17 +141,21 @@ def emit(self, record):
 logger = logging.LoggerAdapter(log, {'OCF_RESOURCE_INSTANCE': OCF_RESOURCE_INSTANCE})
 
 
+_exit_reason_set = False
```

Comment:
Wouldnt it be better to have this as a parameter in the Agent class?

Reply:
yeah, that's cleaner. I'll do that.

Reply:
Ah... actually, that won't work. The idea is to catch the case where the agent code already set the exit reason, so that catching the exception doesn't overwrite that. But since `ocf_exit_reason` is a free function, it needs to record having been called itself.